v3/src/v2/import-v2-document.ts

```diff
@@ -32,7 +32,7 @@ export async function importV2Document(v2Document: CodapV2Document) {
       if (row && tile) {
         const info = getTileComponentInfo(tile.content.type)
         if (info) {
-          const { left, top, width, height } = v2Component.layout
+          const { left = 0, top = 0, width, height } = v2Component.layout
```

Comment:
Out of curiosity, did you happen to check how v2 handles this situation? It seems odd to me that these values can be undefined.

Reply:
I didn't dig into the code for v2 about this, but I did check that it makes the Markov plugin appear at about 0,0.

The main problem this is solving is that `isFreeTileInRowOptions` in `free-tile-row.ts` checks for `x` and `y` to determine if an object is an `IFreeTileInRowOptions`. `FreeTileRow.insertTile` uses this function to make sure the options passed to it are actually `IFreeTileInRowOptions`, and if they aren't it uses an empty object for the options, which results in `height` and `width` of `undefined` (even if these values are included in the options that were deemed the wrong type). Setting `x` and `y` to 0 as the default here ensures that the options passed to `content?.insertTileSnapshotInRow` on line 39 of `import-v2-document.ts` will pass the `isFreeTileInRowOptions` check, so their `height` and `width` won't be ignored. I figured this was the safest, simplest way to avoid this problem, but maybe there's a better solution?

Reply:
I think you're right that this is probably the best way to handle it. I'm just surprised that those properties are written out as `undefined`. The reason I wrote the free tile check that way is because I assumed that surely every CODAP component must save its position. If v2 handles such components by placing them in the top left, then defaulting to zero seems like the right approach.